src/sonic-yang-models/yang-models/sonic-system-ldap.yang

```diff
@@ -0,0 +1,128 @@
+module sonic-system-ldap {
+    yang-version 1.1;
+    namespace "http://github.com/Azure/sonic-system-ldap";
+    prefix ssys;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-port {
+        prefix port;
+    }
+
+    import sonic-portchannel {
+        prefix lag;
+    }
+
+    import sonic-loopback-interface {
+        prefix loopback;
+    }
+
+    import sonic-interface {
+        prefix interface;
+    }
+
+    import sonic-mgmt_port {
+        prefix mgmt-port;
+    }
+
+    description "LDAP YANG Module for SONiC OS";
+
+    revision 2023-10-01 {
+        description "First Revision";
+    }
+
+    container sonic-system-ldap {
+
+        container LDAP_SERVER {
+            list LDAP_SERVER_LIST {
+                max-elements 8;
+                key "ipaddress";
+
+                leaf ipaddress {
+                    type inet:host;
+                    description
+                        "LDAP server's Domain name or IP address (IPv4 or IPv6)";
+                }
+
+                leaf priority {
+                    default 1;
+                    type uint8 {
+                        range "1..8" {
+                            error-message "LDAP server priority must be 1..8";
+                        }
+                    }
+                    description "Server priority";
+                }
+            }
+        }
+
+        container LDAP {
+
+            container global {
+
+
+                leaf bind_dn {

```

Comment:
In the ldap.conf file do these  correspond to the bind_dn --> binddn, bind_password --> bindpw, bind_timeout --> bind_timelimit, timeout --> timelimit, base_dn --> base  ?

Reply:
correct

Reply:
Why not use the same names as in nslcd.conf? It will be less of burden of mapping the nslcd.conf names to users.

Reply:
for now, we decided to remain with the current names since they are human readable as well.

Reply:
Aren't the nslcd.conf names supposed to be human readable? It is a flat text file with descriptive comments.

Reply:
yes and no,
from Linux POV yes, but we are using an special sonic script `sonic-cli-gen`which is generating the CLI from the yang file
So for now, we decided to saved the code as is regarding naming,
If you find any functional bug, pls share.